engine/src/cmd/collide2/Ice/IceTypes.h

```diff
@@ -45,38 +46,38 @@
 	#define null				0														//!< our own NULL pointer
 
 	// Custom types used in ICE
-	typedef signed char			sbyte;		//!< sizeof(sbyte)	must be 1
-	typedef unsigned char		ubyte;		//!< sizeof(ubyte)	must be 1
-	typedef signed short		sword;		//!< sizeof(sword)	must be 2
-	typedef unsigned short		uword;		//!< sizeof(uword)	must be 2
-	typedef signed int			sdword;		//!< sizeof(sdword)	must be 4
-	typedef unsigned int		udword;		//!< sizeof(udword)	must be 4
-	typedef int64		sqword;		//!< sizeof(sqword)	must be 8
-	typedef uint64	uqword;		//!< sizeof(uqword)	must be 8
-	typedef float				float32;	//!< sizeof(float32)	must be 4
-	typedef double				float64;	//!< sizeof(float64)	must be 4
-
-	// ICE_COMPILE_TIME_ASSERT(sizeof(bool)==1);	// ...otherwise things might fail with VC++ 4.2 !
-	ICE_COMPILE_TIME_ASSERT(sizeof(ubyte)==1);
-	ICE_COMPILE_TIME_ASSERT(sizeof(sbyte)==1);
-	ICE_COMPILE_TIME_ASSERT(sizeof(sword)==2);
-	ICE_COMPILE_TIME_ASSERT(sizeof(uword)==2);
-	ICE_COMPILE_TIME_ASSERT(sizeof(udword)==4);
-	ICE_COMPILE_TIME_ASSERT(sizeof(sdword)==4);
-	ICE_COMPILE_TIME_ASSERT(sizeof(uqword)==8);
-	ICE_COMPILE_TIME_ASSERT(sizeof(sqword)==8);
+    typedef int8_t              ice_sbyte;      //!< sizeof(ice_sbyte)      must be 1
+    typedef uint8_t             ice_ubyte;      //!< sizeof(ice_ubyte)      must be 1
+    typedef int16_t             ice_sword;      //!< sizeof(ice_sword)      must be 2
+    typedef uint16_t            ice_uword;      //!< sizeof(ice_uword)      must be 2
+    typedef int32_t             ice_sdword;     //!< sizeof(ice_sdword)     must be 4
+    typedef uint32_t            ice_udword;     //!< sizeof(ice_udword)     must be 4
+    typedef int64_t             ice_sqword;     //!< sizeof(ice_sqword)	    must be 8
+    typedef uint64_t            ice_uqword;     //!< sizeof(ice_uqword)     must be 8
```

Comment:
Question: what is relevant to convey here?

I don't want to create churn for the sake of churn so before any change is made (if any) it'd like to have a proper discussion on that aspect.

Personally I think the bit sizes are more important than the word/dword/qword information as it's more meaningful. If that's the case, then `ice_ubyte` should be `ice_uint8`. If there is a necessary meaning to using word/dword/qword then it's fine AS-IS; but I want to make sure there's good reason. Most have moved away from the word/dword/qword nomenclature AFAICT; so do we need to stick with it too? Or should we move on?

Again, raising the question at the moment, not necessary asking for an immediate change.

Reply:
Unfortunately, I agree. Word is 32 bit(?). Dword and qword are terms I'd have to check.
Honestly, ~~with the introduction of bit size variables~~, why are we using typedefs?
The codebase has a fascination with adding complexity, which just makes it more and more difficult to add stuff.

Reply:
@BenjamenMeyer That's a good point. `ice_uint8` is probably a lot better, actually. And that change should be easy enough to make. I guess I'll go ahead and make that correction.

Reply:
sounds like there's some agreement.

@royfalk yes, `WORD` is 16 bits, `DWORD` = Double `WORD`, 32-bits; `QWORD` = Quad `WORD` so 64-bits; but that assumes you know that a `WORD` is two `bytes`. Typically WORD/DWORD/QWORD are all unsigned values too. https://en.wikipedia.org/wiki/Word_(computer_architecture)#Size_families has some good info on this whole thing, which while making some of it clearer will also make things a lot more confusing.

Reply:
OK. I've gone ahead and changed the type names to `ice_uint8`, `ice_sint8` (signed 8-bit integer for the ICE subsystem), etc. But should we get rid of the typedefs, and just go ahead and use `uint8_t`, etc. directly?

Reply:
@stephengtuggy does it add value? if not, change it.

Reply:
👍🏽 I'll change it.

Reply:
Done